app/services/request-preview.js

```diff
@@ -0,0 +1,45 @@
+import Service, { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { computed } from '@ember/object';
+import { reads, or } from '@ember/object/computed';
+import config from 'travis/config/environment';
+
+export default Service.extend({
+  store: service(),
+
+  loaded: false,
+  loading: reads('loadConfigs.isRunning'),
+  config: reads('record.config'),
+  rawConfigs: reads('record.rawConfigs'),
+  jobConfigs: reads('record.jobConfigs'),
+  errorMessages: computed(() => []),
+  messages: or('record.messages', 'errorMessages'),
+
+  loadConfigs: task(function* (data, debounce) {
+    if (debounce) {
+      const { searchDebounceRate } = config.intervals;
+      yield timeout(searchDebounceRate);
+    }
+
+    try {
+      data.repo = yield this.store.findRecord('repo', data.repo.get('id'));
+      const record = this.store.createRecord('request-preview', data);
+      yield record.save();
+      this.setProperties({ record: record, loaded: true });
```

Comment:
hmmmm, i believe we've tried this before, or do i misremember?

i think the issue is that the `request/configs` component needs to be able to `reset` this property (i.e. set it to `false`) when the user clicks `cancel` and closes the component. the property previously was defined on the component, rather than the service in order to not keep this state here ... but then after some discussion with Sophie (if i remember correctly?) we've decided to move it here.

Reply:
fwiw there _is_ a way to reset the state on the service, and this is now resolved here: https://github.com/travis-ci/travis-web/commit/f6d98997905ecbabb29a1fc575bde48166d7aa5a